espflash/src/error.rs

```diff
@@ -210,6 +210,9 @@ pub enum Error {
 
     #[error("Failed to parse partition table")]
     Partition(#[from] esp_idf_part::Error),
+
+    #[error("Invalid response length")]
```

Comment:
Maybe it would be nice to include some additional information here, i.e. what about the response was invalid

Reply:
I'm not sure what else could be added here from reading esptool docs/code

Reply:
Well, a lot of the response parsing can probably be simplified. Looking at the code I'm not really sure this error is really necessary. e.g. currently we have a bunch of instances of code like this:

```rust
res.get(0..4)
  .ok_or(Error::InvalidResponse)?
  .try_into()
  .map_err(|_| Error::InvalidResponse)?,
```

We have already verified the length of the response and know that we have enough bytes, so we can simply index into the bytes and remove the first error check altogether:

```rust
res.get[0..4]
  .try_into()
  .map_err(|_| Error::InvalidResponse)?,
```

From here, it's not clear to my how `try_into` can actually fail, if we already know we have the correct number of bytes 🤔

Reply:
> the expected length and actual length would probably be useful pieces of information to include in the error.

It seems like different revisions could have different size of response (at least it's the case with S2), I don't see a good way how to do this.

Reply:
Yeah not sure what's going on here. I have a v0.0 and running `board-info` works as expected but your script is failing for me. Will look into it tomorrow.

Reply:
I stumbled across [this comment](https://github.com/espressif/esptool/blob/master/esptool/loader.py#L232) in `esptool.py`, adding `--no-stub` to your script it's now working for me. Not sure how it works for other chips without this (I'm obviously missing something here), but FYI.

Reply:
I think I got it - the reason here was, I didn't rebase my branch to use C stubs so that's why it worked for me locally and failed in CI. C stub gives different length of response for S2 compared to Rut stub.